### DISCUSSION TITLE
Why `usePrepareContractWrite` prepare config if `enabled` is set to `false`

### DISCUSSION THREAD

**Discussion by jeromevvb** [wevm/wagmi#2094] [Q&A]
Hi,

This my usage of `usePrepareContractWrite`

```

const prepareContract = usePrepareContractWrite({
    address: paymentTokenContractAddress as `0x${string}`,
    abi: erc20ABI,
    functionName: 'approve',
    args: [campaignContractAddress as `0x${string}`, _funds],
    enabled: false
  });

```

When i log `prepareContract.config`, i see:
<img width="638" alt="Screenshot 2023-03-29 at 8 54 06 PM" src="https://user-images.githubusercontent.com/7522096/228542116-5443d494-8e05-4bf3-88c7-24d5792d72e9.png">

Why mode is set to `prepared`?

**Comment by SaadNiaziDev** [comment]
Not sure but when u console log , its like variable defination being run!. (in ur case ) 
What if u use as react query structure like const {config} = usePrepareContractWrite({....});, could u try in this way ? There is no reason config will run without explicitly calling it when enabled set to false.

**Comment by jeromevvb** [comment]
Thank you your answer but it does not change anything
I still don't understand why enabled does not behave as expected, the docs is very unclear about it.

**Comment by invalidBan** [comment]
@tmm @jxom Please help. 

I am having the same problem. I expected the `write` function returned by `useContractWrite` to be `undefined` by setting `enabled` to false in `usePrepareContractWrite`. How can the consumer of `useContractWrite` know if the write function is disabled?

**Reply by invalidBan** [reply]
OKay, it seems after updating wagmi to latest version, the problem goes away.

**Comment by salieflewis** [comment]
It's on my list to provide a reproducible example for this, but this issue still seems to be present as far as I can tell. Currently using wagmi @ 1.3.10.

A note in the docs suggests that "the `write` function will be undefined if the config has not been prepared" so this is what I'm using as the barometer for whether or not a transaction is prepared. 

The mode returned from the config object remains unreliable.

**Reply by salieflewis** [reply]
Following up after spending some time with this. I think the `mode` returned by the config is hardcoded as `prepared`, and me anticipating an `unprepared` response if the prepare hook is not enabled was a misunderstanding of the API.
